mac/deploy_mac.sh

```diff
@@ -67,18 +67,28 @@ build_app()
 
 build_installer_image()
 {
-    # Install dmgbuild (for the current user), this is required to build the installer image
-    python -m ensurepip --user --default-pip
-    python -m pip install --user dmgbuild==1.4.2
-    local dmgbuild_bin="$(python -c 'import site; print(site.USER_BASE)')/bin/dmgbuild"
-
+    # this needs brew to be installed first
+    # install create-dmg
+    brew install create-dmg
```

Comment:
Do we want to rely on brew?

Reply:
Is there a better alternative?
I think we should keep external package manager usage to a minimum, but using one is certainly better than download binaries (and their dependencies!) manually all over the place.

If we do use brew, we should update #2277 accordingly.

Can you do version pinning here?

Reply:
Not sure how. I read it should work with @version but the ci failed

Reply:
Looks like there is no easy way. :(

This seems to work but sounds strange to me. It also downloads a rather huge git repo to do that.
```
root@cc5850aba45c:/home/linuxbrew# brew extract --version=1.0.9 create-dmg homebrew/cask
Warning: extract is a developer command, so
Homebrew's developer mode has been automatically turned on.
To turn developer mode off, run brew developer off

==> Tapping homebrew/cask
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 618845, done.
remote: Total 618845 (delta 0), reused 0 (delta 0), pack-reused 618845
Receiving objects: 100% (618845/618845), 282.97 MiB | 7.00 MiB/s, done.
Resolving deltas: 100% (437655/437655), done.
Tapped 3959 casks (4,030 files, 302.8MB).
==> Searching repository history
==> Writing formula for create-dmg from revision 61daf01 to:
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask/Formula/create-dmg@1.0.9.rb
root@cc5850aba45c:/home/linuxbrew# brew install /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask/Formula/create-dmg@1.0.9.rb
Error: Failed to load cask: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask/Formula/create-dmg@1.0.9.rb
Cask 'create-dmg@1.0.9' is unreadable: wrong constant name #<Class:0x000000000270c540>
Warning: Treating /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask/Formula/create-dmg@1.0.9.rb as a formula.
==> Downloading https://github.com/create-dmg/create-dmg/archive/v1.0.9.tar.gz
==> Downloading from https://codeload.github.com/create-dmg/create-dmg/tar.gz/v1.0.9
#=#=#                                                                         
==> Installing create-dmg@1.0.9 from homebrew/cask
==> make install prefix=/home/linuxbrew/.linuxbrew/Cellar/create-dmg@1.0.9/1.0.9
🍺  /home/linuxbrew/.linuxbrew/Cellar/create-dmg@1.0.9/1.0.9: 12 files, 61.4KB, built in 1 second
==> Running `brew cleanup create-dmg@1.0.9`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

Reply:
Didn't seem to work on the CI. I've reverted it.

Reply:
Argh. You need an additional install step